docker/src/python/group-sync/group-sync.py

```diff
@@ -0,0 +1,51 @@
+import json
+import subprocess
+import os
+import sys
+
+def add_users_to_group(group_name, namespace):
+    # Run the `oc get` command, capture the JSON output, and load the data
+    oc_get_output = subprocess.run(["oc", "get", "-o", "json", "rolebinding", "edit", "-n", namespace], capture_output=True, text=True)
+
+    # Check if the command was successful and load the JSON data
+    if oc_get_output.returncode == 0:
+        data = json.loads(oc_get_output.stdout)
+
+        # Extract subjects
+        subjects = data.get("subjects", [])
+
+        try:
+            # Try to get the current members of the OCP group
+            oc_get_output = subprocess.run(["oc", "get", "group", group_name, "-o", "json"], capture_output=True, text=True)
+            group_info = json.loads(oc_get_output.stdout)
+            current_members = group_info.get("users", [])
+        except json.JSONDecodeError as e:
+            # Handle the case where JSON decoding fails (an empty group)
+            print(f"Error decoding JSON: {e}")
+            current_members = []
+
+        # Add usernames to the OCP group if not already members
+        for subject in subjects:
+            username = subject.get("name")
+            
+            if username not in current_members:
+                # Add the username to the OCP group
+                subprocess.run(["oc", "adm", "groups", "add-users", group_name, username])
```

Comment:
Have you tested the performance to add hundreds users to a group? Is it possible to collect the users that need to be added to group and then add them all at once, in order to save api calls.

Reply:
The performance is very good, the script runs to completion in about 30 seeconds or less, even with adding every student